plugins/jobs/src/js/components/form/RunConfigFormSection.tsx

```diff
@@ -0,0 +1,424 @@
+import * as React from "react";
+import { Trans } from "@lingui/macro";
+
+import AddButton from "#SRC/js/components/form/AddButton";
+import DeleteRowButton from "#SRC/js/components/form/DeleteRowButton";
+import FieldAutofocus from "#SRC/js/components/form/FieldAutofocus";
+import FieldHelp from "#SRC/js/components/form/FieldHelp";
+import FieldInput from "#SRC/js/components/form/FieldInput";
+import FieldLabel from "#SRC/js/components/form/FieldLabel";
+import FormGroup from "#SRC/js/components/form/FormGroup";
+import FormGroupContainer from "#SRC/js/components/form/FormGroupContainer";
+import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
+import FormGroupHeadingContent from "#SRC/js/components/form/FormGroupHeadingContent";
+import FormRow from "#SRC/js/components/form/FormRow";
+import InfoTooltipIcon from "#SRC/js/components/form/InfoTooltipIcon";
+import classNames from "classnames";
+import { FormOutput, FormError, RestartPolicy } from "./helpers/JobFormData";
+import { Tooltip } from "reactjs-components";
+
+interface RunConfigSectionProps {
+  formData: FormOutput;
+  errors: FormError[];
+  showErrors: boolean;
+  onRemoveItem: (path: string, index: number) => void;
+  onAddItem: (path: string) => void;
+}
+
+class RunConfigFormSection extends React.Component<RunConfigSectionProps> {
+  constructor(props: RunConfigSectionProps) {
+    super(props);
+  }
+
+  render() {
+    const { formData, onAddItem, onRemoveItem } = this.props;
+    const showErrors = false;
+    const labels = formData.labels || [];
+    const artifacts = formData.artifacts || [];
+
+    try {
+      return (
+        <div className="form-section">
+          {/*
+
+            RUN CONFIGURATION
```

Comment:
I like that you introduced these comments for better readability. I sense that you, too, have the feeling that this components render function is just too big. Have you considered breaking it up into multiple smaller ones containing the sections? then we could easily change the order and have named things in there.

Reply:
TL;DR this is intentional and i'd like to keep it that way.

i like long render functions as long as they're dumb. unless we want to reuse some code i don't feel like breaking up stuff here makes the code more readable or maintainable. the only thing i see as a benefit of splitting up would be the ability to jump to a section with our editors' jump-to-definition-functionality faster, once we found the render-function.

Reply:
Hmm I feel your concerns and I see where you are coming from. Although the render function is huge, I don't have a better Idea. But since this class has only the render function I would feel more comfortable with changing the class to a single function component. Although, this can be addressed at a later point.

Reply:
that's a great idea!